### PR TITLE
Allow the pipeline configuration to filter for specific files

### PR DESCRIPTION
**Pull request by cbeer**

### CODE REVIEW

File: config/settings.yml [hunk 1]
```diff
@@ -29,4 +29,5 @@ import:
         inst_id: stanford
     met_csv:
       directory: museum/records
+      filter: !ruby/regexp '/met_museum_records.csv/'
```
Comment (jcoyne) [inline config/settings.yml]: Is this an exclusionary filter?  If so can we call it `excluding`?

Comment (cbeer) [inline config/settings.yml]: No, it's selecting resources that match.

Comment (jcoyne) [inline config/settings.yml]: Can we call it `only` then?

File: lib/github_importer.rb [hunk 2]
```diff
@@ -12,6 +12,8 @@ def initialize(oauth_token, repo)
   # @yield [HarvestedResource] Gives the harvested resource
   def import(harvest, pipeline)
     gh.contents(repo, path: pipeline.config.directory).each do |resource|
+      next if pipeline.filter && resource.name !~ pipeline.filter
```
Comment (jcoyne) [inline lib/github_importer.rb]: I don't understand how this is resources that match. It looks to me like we skip importing (`next`) if the file matches the filter. Right?

Comment (jcoyne) [inline lib/github_importer.rb]: Doh. I just saw that you were using the not-matching operator.

Comment (cbeer) [inline lib/github_importer.rb]: If there's a filter, and it doesn't match the `pipeline.filter`, we skip it.

File: config/settings.yml (hunk 1 above)
Comment (cbeer) [inline config/settings.yml]: Sure.

File: lib/github_importer.rb [hunk 3]
```diff
@@ -12,6 +12,8 @@ def initialize(oauth_token, repo)
   # @yield [HarvestedResource] Gives the harvested resource
   def import(harvest, pipeline)
     gh.contents(repo, path: pipeline.config.directory).each do |resource|
+      next if pipeline.filter && resource.name !~ pipeline.onlyu
```
Comment (jcoyne) [inline lib/github_importer.rb]: typo

Comment (jcoyne) [inline lib/github_importer.rb]: https://youtu.be/3FygIKsnkCw?t=10s

**Pull request merged**